CorsixTH/Lua/hospital.lua

```diff
@@ -1517,8 +1517,16 @@ function Hospital:initStaff()
       if added_staff then
         local staff = self.world:newEntity(profile.humanoid_class, 2)
         staff:setProfile(profile)
-        -- TODO: Make a somewhat "nicer" placing algorithm.
-        staff:setTile(self.world.map.th:getCameraTile(1))
+
+        local x, y = self.world.map.th:getCameraTile(1)
+        local x2, y2
+        local attempts = 0
+        repeat
+          x2 = x + math.random(-10, 10)
+          y2 = y + math.random(-10, 10)
+          attempts = attempts + 1
+        until self:isInHospital(x2, y2) or attempts > 100
+        staff:setTile(x2, y2)
```

Comment:
So you tried a 100 times and failed, and then put the staff member down to the last randomly selected tile.
Depending on how bad the camera location is, it may even be off-map. [Question: does the `isInHospital` function handle off-map coordinates in case the camera is close to the edge of the map, or does it simply crash?]

One way to handle it would be to make sure the original `x`, `y` pair is in the hospital and use that if 100 tries all fail. A second way is to query the world for a tile that is in the hospital (don't think we have that function yet), in case all attempts fail.